### PR TITLE
Refactor DebugEventType to TrackingEventType

### PR DESCRIPTION
**Pull request by TimvdLippe**
This refactors the enum `DebugEventType` to `TrackingEventType`. In the process, I refactored some duplication from list handling in the eventManager and consequently removed the identical implementations in Eclipse and IntelliJ. This meant that I had to move jfreechart from eclipse/intellij to core.

### CODE REVIEW

File: core/src/nl/tudelft/watchdog/core/logic/event/eventtypes/TrackingEventType.java [hunk 1]
```diff
@@ -6,57 +6,70 @@
  * Enumeration of the different possible debug events that can be fired based on the
  * activities a developer can perform.
  */
-public enum DebugEventType {
+public enum TrackingEventType {
```
Comment (Inventitech) [inline core/src/nl/tudelft/watchdog/core/logic/event/eventtypes/TrackingEventType.java]: I think this is completely analogous to our IntervalType enum, and should thus follow the same naming scheme

File: core/src/nl/tudelft/watchdog/core/logic/event/EventStatisticsBase.java [hunk 2]
```diff
@@ -2,21 +2,29 @@
...
+import org.jfree.data.gantt.TaskSeriesCollection;
+
 /**
  * Contains basic functionality for selecting all events that occurred during a
  * certain debug interval. Dataset creation functionality should be implemented
  * by IDE-specific code in order to avoid IDE-specific dependencies in the core
  * project.
  */
-public abstract class EventStatisticsBase {
+public class EventStatisticsBase {
```
Comment (Inventitech) [inline core/src/nl/tudelft/watchdog/core/logic/event/EventStatisticsBase.java]: This isn't purely a base class anymore

File: eclipse/tests/src/nl/tudelft/watchdog/logic/event/EventManagerTest.java [hunk 3]
```diff
@@ -12,25 +12,25 @@
...
- * Tests for testing the correctness of the {@link DebugEventManager} when events are
+ * Tests for testing the correctness of the {@link TrackingEventManager} when events are
  * added.
  */
 @RunWith(PowerMockRunner.class)
 @PrepareForTest(WatchDogGlobals.class)
 public class EventManagerTest {
 
-	private DebugEventManager debugEventManager;
+	private TrackingEventManager TrackingEventManager;
```
Comment (Inventitech) [inline eclipse/tests/src/nl/tudelft/watchdog/logic/event/EventManagerTest.java]: Camel case?

File: intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorWindowListener.java [hunk 4]
```diff
@@ -13,44 +19,53 @@
```
Comment (Inventitech) [inline intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorWindowListener.java]: Why these changes?

File: intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/IntelliJMarkupModelListener.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package nl.tudelft.watchdog.intellij.logic.ui.listeners;
```
Comment (Inventitech) [inline intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/IntelliJMarkupModelListener.java]: This seems to be a first step toward implementing ASAT tracking functionality? Different PR?

File: core/src/nl/tudelft/watchdog/core/logic/event/eventtypes/TrackingEventType.java (hunk 1 above)
Comment (TimvdLippe) [reply core/src/nl/tudelft/watchdog/core/logic/event/eventtypes/TrackingEventType.java]: I am  not sure which naming scheme you mean. Could you give a suggestion for a better name?

File: eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/WorkbenchListener.java [hunk 6]
```diff
@@ -66,6 +79,52 @@ public void attachListeners() {
 				workbench.getDisplay());
```
[the remark this answers is not in the document]
Comment (TimvdLippe) [reply eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/WorkbenchListener.java]: I did not touch this line and it is part of the previous line, I think this should remain as-is? 
image

File: intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorWindowListener.java (hunk 4 above)
Comment (TimvdLippe) [reply intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorWindowListener.java]: They were part of my preparation for the ASAT functionality and to make it more idiomatic IntelliJ code. Will put them in a separate PR :+1:

File: eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/InitializationManager.java [hunk 7]
```diff
@@ -32,7 +32,7 @@
 	private final PersisterBase statisticsPersister;
 
 	private final WatchDogEventManager watchDogEventManager;
```
[the remark this answers is not in the document]
Comment (TimvdLippe) [reply eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/InitializationManager.java]: Add comments to clarify.

File: intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorListener.java [hunk 8]
```diff
@@ -86,29 +85,29 @@ public void caretAdded(CaretEvent e) {
...
-		editor.getScrollingModel().addVisibleAreaListener(visibleAreaListener);
+        editor.getScrollingModel().addVisibleAreaListener(visibleAreaListener);
 
-	}
+    }
 
-	@Override
-	public void dispose() {
+    /** Removes all listeners registered with this editor. */
+    public void removeListeners() {
```
Comment (Inventitech) [inline intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorListener.java]: When and how is this called now?

File: intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java [hunk 9]
```diff
@@ -32,7 +32,13 @@
     private final Persister toTransferPersister;
     private final Persister statisticsPersister;
 
+    /**
```
Comment (Inventitech) [inline intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java]: Don't the comments suggest that a more appropriate naming of these three should be done? Also, what's the difference between the WatchDogEventManager and the intervalManager now?

File: eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/WorkbenchListener.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Inventitech) [reply eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/WorkbenchListener.java]: Yap

File: intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java (hunk 9 above)
Comment (TimvdLippe) [reply intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java]: I think this is mostly an artifact of the usage of `*Manager`. In general, "manager" is very generic and hard to grasp. The distinction between watchdog events, intervals and debug events is ambiguous as well. Rather than resolve that in this PR, I would like to discuss with you better names and architecture to resolve this ambiguation. WDYT?

File: intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorListener.java (hunk 8 above)
Comment (TimvdLippe) [reply intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorListener.java]: Hm, I have reverted these changes, but it appears to be confusing with my whitespace changes. Apparently my editor is configured to use spaces, but the files use tabs. In https://github.com/TestRoots/watchdog/pull/274/files?w=1#diff-1b718277d0fe136c9530faa96dd1367e you can see that there were only whitespace changes in this file.

File: intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java (hunk 9 above)
Comment (Inventitech) [reply intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java]: I agree on the manager part, but this will be hard to change.

What about the following?

WatchDogEvent -> IDEEvent
TrackingEvent -> WatchDogEvent
Interval -> WatchDogInterval

If you still want to do this as part of a separate PR (or have different ideas on the rename), that is more than fine with me.

Comment (Inventitech) [reply intellij/src/nl/tudelft/watchdog/intellij/logic/InitializationManager.java]: Merging because this is now part of #275. Thx, Tim!
